### PR TITLE
Add additional docs for uncooperative ctor deprecation

### LINKED ISSUE #9488: [prerelease] Improve cooperate constructor warning
When trying to look at the last couple of issues in #9415, I saw this (from pytest-flake8):

```
 PytestDeprecationWarning: <class 'pytest_flake8.Flake8Item'> is not using a cooperative constructor and only takes {'fspath', 'parent'}
```

It took me a bit of guessing to know that the warning probably wants me to add `**kwargs` and pass it on. This was done as part of #8447, but it seems to me like it should have a section in the [deprecation docs](https://docs.pytest.org/en/latest/deprecations.html) we could link to, as well as a dedicated changelog entry.

### PR DESCRIPTION
**Pull request by The-Compiler**
Fixes #9488

Probably still needs test adjustments, but I'd like some feedback from @RonnyPfannschmidt if this is the way to go.

### CODE REVIEW

File: doc/en/example/nonpython/conftest.py [hunk 1]
```diff
@@ -18,8 +18,8 @@ def collect(self):
 
 
 class YamlItem(pytest.Item):
-    def __init__(self, name, parent, spec):
-        super().__init__(name, parent)
+    def __init__(self, spec, **kwargs):
+        super().__init__(**kwargs)
```
[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply doc/en/example/nonpython/conftest.py:22]: the warning was introduced to handle broken code

if we introduce it like this, then we should also take a look at standardizing to kwonly args so we can at some point deprecate positional args to nodes

File: doc/en/deprecations.rst [hunk 2]
```diff
@@ -191,6 +195,40 @@ Instead, a separate collector node should be used, which collects the item. See
...
+        def __init__(self, name, parent, additional_arg):
+            super().__init__(name, parent)
+            self.additional_arg = additional_arg
+
+should be turned into:
+
+.. code-block:: python
+
+    class CustomItem(pytest.Item):
+        def __init__(self, additional_arg, **kwargs):
```
Comment (RonnyPfannschmidt) [inline doc/en/deprecations.rst:220]:
```suggestion
        def __init__(self, *,  additional_arg, **kwargs):
```

**Review by RonnyPfannschmidt — APPROVED** [review]
approving to ease  process before im off to take the toddler , lets add the `*` to make it kwonly tho

File: doc/en/deprecations.rst [hunk 3]
```diff
@@ -191,6 +195,40 @@ Instead, a separate collector node should be used, which collects the item. See
...
+``__init__`` method, they should take ``**kwargs``. Thus,
+
+.. code-block:: python
+
+    class CustomItem(pytest.Item):
+        def __init__(self, name, parent, additional_arg):
+            super().__init__(name, parent)
+            self.additional_arg = additional_arg
+
+should be turned into:
```
[the remark this answers is not in the document]
Comment (The-Compiler) [reply doc/en/deprecations.rst:215]: Disagree, as the code snippet is in the middle of a sentence - but if you feel strongly about this, I can change it, no biggie.

File: doc/en/deprecations.rst (hunk 2 above)
Comment (The-Compiler) [reply doc/en/deprecations.rst:220]: Hmm, I'm not sure about this honestly - while I agree it's a good thing in general, in the context of this deprecation warning, it seems more confusing than useful, no?

File: doc/en/deprecations.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicoddemus) [reply doc/en/deprecations.rst:215]: No that's fine. 😁

File: doc/en/deprecations.rst (hunk 2 above)
Comment (bluetech) [reply doc/en/deprecations.rst:220]: I agree with @RonnyPfannschmidt, I think we should encourage using kwarg-only in the ctors.

File: doc/en/example/nonpython/conftest.py [hunk 4]
```diff
@@ -18,8 +18,8 @@ def collect(self):
 
 
 class YamlItem(pytest.Item):
-    def __init__(self, name, parent, spec):
-        super().__init__(name, parent)
+    def __init__(self, spec, **kwargs):
```
Comment (bluetech) [inline doc/en/example/nonpython/conftest.py:21]: Following discussion above:

```suggestion
    def __init__(self, *, spec, **kwargs):
```

**Review by bluetech — APPROVED** [review]
LGTM!

**Comment by The-Compiler** [comment]
Apologies for the delay, I forgot about this. Updated now.

**Pull request merged**